File: c++/1904.cpp
#include <iostream>

using namespace std;

int dp[1000001];

int main() {
  freopen("example.txt", "r", stdin);

  int N;
  cin >> N;

  dp[1] = 1;
  dp[2] = 2;

  for (int i = 3; i <= N; i++) {
    dp[i] = (dp[i - 1] + dp[i - 2]) % 15746;
  }

  cout << dp[N];

  return 0;
}

File: js/1904.js
const fs = require('fs');

const N = +fs.readFileSync('example.txt').toString().trim();

const dp = Array(N + 1).fill(0);

dp[1] = 1;
dp[2] = 2;

for (let i = 3; i <= N; i++) {
  dp[i] = (dp[i - 1] + dp[i - 2]) % 15746;
}

console.log(dp[N]);
